examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/BarcodeFragment.kt

```diff
@@ -199,14 +196,14 @@ class BarcodeFragment : Fragment() {
             Toast.makeText(requireContext(), "Unrecognized QR Code", Toast.LENGTH_SHORT).show()
         }
         FragmentUtil.getHost(this@BarcodeFragment, Callback::class.java)
-            ?.onCHIPDeviceInfoReceived(CHIPDeviceInfo.fromSetupPayload(payload, isShortDiscriminator))
+            ?.onCHIPDeviceInfoReceived(CHIPDeviceInfo.fromSetupPayload(payload))
     }
 
     private fun handleScannedQrCode(barcode: Barcode) {
         Handler(Looper.getMainLooper()).post {
-            lateinit var payload: SetupPayload
+            lateinit var payload: OnboardingPayload
             try {
-                payload = SetupPayloadParser().parseQrCode(barcode.displayValue)
+                payload = barcode.displayValue?.let { OnboardingPayloadParser().parseQrCode(it) } ?: return@post
```

Comment:
?: Elvis operator is used to provide a default value or an alternative action when the expression before it is null. In this case, if displayValue is null, the code will immediately return from the current function or lambda, indicated by return@post.

Shall we assert or throw exception if displayValue is null?

Reply:
https://github.com/project-chip/connectedhomeip/blob/237784456a22ac51fe00c55179e27c8e58239974/examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/BarcodeFragment.kt#L213

As you can see in this code, even if an exception occurs, it is terminated with return@post below. Therefore, I think it is better to return than to throw an exception at this location.